added HOD model to the hod.py script

yamada-yuka0322 opened the pull request:
Added HOD model classes,  and function to populate galaxies

yamada-yuka0322 committed:
added HOD model to the HOD.py script

changhoonhahn commented on src/pfs-sims/hod.py:
```diff
@@ -4,3 +4,153 @@
 
 
 '''
+#HOD model
```
Remove any packages that aren't used (e.g. scipy.special.erfc)

changhoonhahn commented on src/pfs-sims/hod.py:
```diff
@@ -4,3 +4,153 @@
+        # Call the parent constructor
+        super(CustomCentrals, self).__init__(gal_type='centrals', threshold=threshold, upper_occupation_bound=1, **kwargs)
+
+        # Define custom parameters
```
The HOD parameters should be an input. Not a default parameter. Also clean up any unnecessary comments.

changhoonhahn commented on src/pfs-sims/hod.py:
```diff
@@ -4,3 +4,153 @@
+            'gamma' : 7.06
+        }
+
+    def mean_occupation(self, **kwargs):
```
For any function, document the input and output variables.

changhoonhahn commented on src/pfs-sims/hod.py:
```diff
@@ -4,3 +4,153 @@
+        return Ac/(2*math.pi)**0.5/sigma_M * np.exp(-(np.log10(halo_mass) - logMc)**2/2/sigma_M**2)
+
+#Satellite galaxy occupation
+class CustomSatellites(OccupationComponent):
```
Document what the class does.

changhoonhahn commented on src/pfs-sims/hod.py:
```diff
@@ -4,3 +4,153 @@
+        self._galprop_dtypes_to_allocate = np.dtype([('vx', 'f8'),('vy', 'f8'), ('vz', 'f8')])
+        self.list_of_haloprops_needed = ['halo_vx', 'halo_vy', 'halo_vz', 'halo_sigmav']
+
+    def assign_velocity(self, **kwargs):
```
Specify input parameters rather than setting them all as kwargs.

changhoonhahn reviewed (commented):
Please address the issues in the pull request.

changhoonhahn commented:
Some additional comments and suggestions @yamada-yuka0322: 
- Please document every function and class objects. Provide descriptive explanations of what they do. Provide a clear list and explanations of the input and outputs of the functions. You are writing this for others to use. Everything has to be clear.
- You are currently assuming that the halo catalog will be read from a file. The reading of the halo catalog will be done in another module (`halo`). So you should assume that a "halo" class object will be passed as input.

yamada-yuka0322 committed:
added explanation and modified hod.py classes to take HOD parameters as inputs

yamada-yuka0322 committed:
Merge branch 'pfs-cosmo:main' into yuka

changhoonhahn committed:
revisions to halo.py

Here's a revised version of the halos.py. The main modifications are: 
- We want the catalog attributes (redshift, Lbox, particle_mass) separated from the catalog itself (position, velocity etc). I've also set the halo catalog to be an `astropy.table.Table` object, which is better organized than a dictionary. 
- I've also moved `load_Abacus` as a separate function that reads in the Abacus halos as a `Halos` object. 

There are still details that need to be filled in so go through them when you have the chance.

changhoonhahn committed:
Update hod.py

I cleaned up the imports. It's good practice to keep dependencies minimal (i.e. remove any you're not using and group them)

A couple of major comments: 
- `CustomCentral`: You're using code that's designed to include luminosity information. That is *not* what we want. At the moment it's unclear what that's doing, which means it could be doing something we don't want. 
- `CustomSatellite`: Same thing. You cannot just copy and paste code and not understand what args and kwargs you're passing are doing.
